### PR TITLE
[RFR] Exporter Istex Query

### PR DESCRIPTION
**Pull request by QuanticPotatoes**

### CODE REVIEW

File: src/api/exporters/exportIstex.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    .pipe(ezs('jsonify'));
+
+exporter.extension = 'json';
+exporter.mimeType = 'application/json';
+exporter.type = 'file';
+exporter.label = 'istex';
+
+export default exporter;
+
+/*
```
Comment (djhi) [inline src/api/exporters/exportIstex.js]: Don't keep commented out code

File: src/api/exporters/ezsLocals.js [hunk 2]
```diff
@@ -20,9 +19,9 @@ export function filterVersions(data, feed) {
 export function filterContributions(data, feed) {
     if (data && data.contributions) {
         const fieldsToIgnore = data.contributions
-            .filter(({ status }) => status !== VALIDATED)
-            .map(({ fieldName }) => fieldName)
-            .concat('contributions', 'contributionCount');
+      .filter(({ status }) => status !== VALIDATED)
```
Comment (djhi) [inline src/api/exporters/ezsLocals.js]: weird indentation

File: src/api/exporters/ezsLocals.js [hunk 3]
```diff
@@ -49,7 +48,7 @@ export function linkDataset(data, feed) {
         data.dataset = uri;
         data['@context'].dataset = {
             '@id': scheme,
-//             '@type': 'https://www.w3.org/TR/xmlschema-2/#anyURI',
```
Comment (djhi) [inline src/api/exporters/ezsLocals.js]: commented out code

File: src/api/exporters/ezsLocals.js [hunk 4]
```diff
@@ -61,9 +60,12 @@ export function JSONLDObject(data, feed) {
     } else {
         const output = {};
 
-        if (data.uri.indexOf('http://') !== 0 &&
-            data.uri.indexOf('https://') !== 0) {
-            data.uri = path.normalize(this.getParam('hostname', 'http://lod.istex.fr/').concat(data.uri));
+        if (
+      data.uri.indexOf('http://') !== 0 && data.uri.indexOf('https://') !== 0
```
Comment (djhi) [inline src/api/exporters/ezsLocals.js]: format: weird indentation

File: src/api/exporters/ezsLocals.js [hunk 5]
```diff
@@ -88,18 +90,33 @@ export function JSONLDObject(data, feed) {
...
 }
 
+export function extractIstexQuery(data, feed) {
+    if (this.isLast()) {
+        feed.close();
+    } else {
+        const fields = this.getParam('fields', {});
+
+        fields
+      .filter(field => field.format && field.format.name === 'istex')
```
Comment (djhi) [inline src/api/exporters/ezsLocals.js]: same

File: src/api/exporters/ezsLocals.js [hunk 6]
```diff
@@ -88,18 +90,33 @@ export function JSONLDObject(data, feed) {
...
+      .forEach((field) => {
+          const propertyName = field.name;
+          console.log(data[propertyName]);
+      });
+        feed.send(data);
```
Comment (djhi) [inline src/api/exporters/ezsLocals.js]: here too

File: src/api/exporters/ezsLocals.js [hunk 7]
```diff
@@ -88,18 +90,33 @@ export function JSONLDObject(data, feed) {
...
         feed.close();
     } else {
-        jsonld.toRDF(data, { format: 'application/nquads' })
-            .then((out) => {
-                feed.send(out);
-            },
-            (err) => {
-                throw err;
-            });
+        jsonld.toRDF(data, { format: 'application/nquads' }).then(
```
Comment (djhi) [inline src/api/exporters/ezsLocals.js]: here again

**Review by djhi — DISMISSED** [review]

**Comment by QuanticPotatoes** [comment]
The weird indentation is generate by my eslint formatter plugin.
I'll see if i can solve this problem.

File: config.json [hunk 8]
```diff
@@ -18,6 +18,7 @@
     "perPage": 10,
     "exporters": [
         "nquads",
+        "istex",
```
Comment (parmentf) [inline config.json]: Shouldn't it be `extendednquads` instead?

**Review by parmentf — CHANGES_REQUESTED** [review]
Adapt the `config.json` to the new name `extendednquads`.

**Review by parmentf — APPROVED** [review]

**Pull request merged**